clients/python/src/model_registry/_client.py

```diff
@@ -45,47 +73,57 @@ def __init__(
         Keyword Args:
             author: Name of the author.
             is_secure: Whether to use a secure connection. Defaults to True.
-            user_token: The PEM-encoded user token as a string. Defaults to content of path on envvar KF_PIPELINES_SA_TOKEN_PATH.
-            custom_ca: Path to the PEM-encoded root certificates as a string. Defaults to path on envvar CERT.
+            user_token: The PEM-encoded user token as a string.
+            user_token_envvar: Environment variable to read the user token from if it's not passed as an arg. Defaults to KF_PIPELINES_SA_TOKEN_PATH.
+            custom_ca: Path to the PEM-encoded root certificates as a string.
+            custom_ca_envvar: Environment variable to read the custom CA from if it's not passed as an arg.
+            log_level: Log level. Defaults to logging.WARNING.
         """
+        logger.setLevel(log_level)
+
         import nest_asyncio
 
+        logger.debug("Setting up reentrant async event loop")
         nest_asyncio.apply()
 
         # TODO: get remaining args from env
         self._author = author
 
-        if not user_token:
+        if not user_token and user_token_envvar:
+            logger.info("Reading user token from %s", user_token_envvar)
+            if user_token_envvar == DEFAULT_USER_TOKEN_ENVVAR:
+                logger.warning(
+                    "Using default user token envvar, if you want to use a different one, pass it as `user_token`"
+                )
```

Comment:
I believe this needs to be moved below, as this warning ("Using ...") will be printed even in the cases that `DEFAULT_USER_TOKEN_ENVVAR` is actually not in the current env. wdyt?

We can _use it_ only if `os.environ.get(user_token_envvar)` is not None.